features/cellular/framework/device/CellularStateMachine.cpp

```diff
@@ -51,26 +53,24 @@ namespace mbed {
 CellularStateMachine::CellularStateMachine(CellularDevice &device, events::EventQueue &queue, CellularNetwork &nw) :
     _cellularDevice(device), _state(STATE_INIT), _next_state(_state), _target_state(_state),
     _event_status_cb(), _network(nw), _queue(queue), _sim_pin(0), _retry_count(0),
-    _event_timeout(-1), _event_id(-1), _plmn(0), _command_success(false),
-    _is_retry(false), _cb_data(), _current_event(CellularDeviceReady), _status(0)
-{
-#if MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY == 0
-    _start_time = 0;
-#else
+#if MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY != 0
     // so that not every device don't start at the exact same time (for example after power outage)
-    _start_time = rand() % (MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY);
+    _start_time(rand() % (MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY)),
```

Comment:
Is else `_start_time(0)` needed here?

Reply:
No, because Chrono durations and time_points are initialised to zero by default. Still, could have it explicit. That would be more consistent with the existing code.

I guess I was pushing back a bit on the long C++98-style initialiser lists on constructors. Since C++11, member initialisation that doesn't depend on constructor parameters shouldn't be there. They should be in default member initialisers (if even necessary).